src/components/Requirements/IncompleteSelfCheck.vue

```diff
@@ -141,50 +138,14 @@ export default defineComponent({
     },
     addExistingCourse(option: string) {
       this.showDropdown = false;
-      const { uniqueID, crseId } = this.selfCheckCourses[option];
-      updateRequirementChoice(uniqueID, choice => ({
-        ...choice,
-        // Since we edit from a self-check requirement,
-        // we know it must be `acknowledgedCheckerWarningOptIn`.
-        acknowledgedCheckerWarningOptIn: Array.from(
-          new Set([...choice.acknowledgedCheckerWarningOptIn, this.subReqId])
-        ),
-        // Keep existing behavior of keeping it connected to at most one requirement.
-        optOut: getRelatedRequirementIdsForCourseOptOut(
-          crseId,
-          this.subReqId,
-          store.state.groupedRequirementFulfillmentReport,
-          store.state.toggleableRequirementChoices,
-          store.state.userRequirementsMap
-        ),
-      }));
+      const { uniqueID } = this.selfCheckCourses[option];
+      addAcknowledgedCheckerWarningOptIn(uniqueID, this.subReqId);
     },
     addNewCourse(course: CornellCourseRosterCourse, season: FirestoreSemesterSeason, year: number) {
       this.showDropdown = false;
       const newCourse = cornellCourseRosterCourseToFirebaseSemesterCourseWithGlobalData(course);
-      addCourseToSemester(
-        year,
-        season,
-        newCourse,
-        // Since the course is new, we know the old choice does not exist.
-        () => ({
-          arbitraryOptIn: {},
-          // Since we edit from a self-check requirement,
-          // we know it must be `acknowledgedCheckerWarningOptIn`.
-          acknowledgedCheckerWarningOptIn: [this.subReqId],
-          // We also need to opt-out of all requirements without warnings,
-          // because the user intention is clear that we only want to bind
-          // the course to this specific requirement.
-          optOut: getRelatedRequirementIdsForCourseOptOut(
-            newCourse.crseId,
-            this.subReqId,
-            store.state.groupedRequirementFulfillmentReport,
-            store.state.toggleableRequirementChoices,
-            store.state.userRequirementsMap
-          ),
-        }),
```

Comment:
Adding the selectable requirement choice is hoisted and moved before the call to `addCourseToSemester`, since I think they should be independent of each other. `getRelatedRequirementIdsForCourseOptOut` is not necessary for the same reason as before.